.github/workflows/publish.yml

```diff
@@ -0,0 +1,87 @@
+name: publish artifacts
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+jobs:
+  build-napi-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: napi build
+        run: yarn install && yarn build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: linux-napi
+          path: ./*.node
```

Comment:
This should be scoped to the directories we intend to build to, rather than the working directory